vector/src/main/java/im/vector/settings/VectorLocale.kt

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.settings
+
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Build
+import android.preference.PreferenceManager
+import android.text.TextUtils
+import android.util.Pair
+import androidx.core.content.edit
+import im.vector.R
+import kotlinx.coroutines.experimental.GlobalScope
+import kotlinx.coroutines.experimental.launch
+import org.matrix.androidsdk.util.Log
+import java.util.*
+
+/**
+ * Object to manage the Locale choice of the user
+ */
+object VectorLocale {
+    private val LOG_TAG = VectorLocale.javaClass.simpleName
+
+    private const val APPLICATION_LOCALE_COUNTRY_KEY = "APPLICATION_LOCALE_COUNTRY_KEY"
+    private const val APPLICATION_LOCALE_VARIANT_KEY = "APPLICATION_LOCALE_VARIANT_KEY"
+    private const val APPLICATION_LOCALE_LANGUAGE_KEY = "APPLICATION_LOCALE_LANGUAGE_KEY"
+
+    private val defaultLocale = Locale("en", "US")
+
+    /**
+     * The supported application languages
+     */
+    var supportedLocales = ArrayList<Locale>()
+        private set
+
+    /**
+     * Provides the current application locale
+     */
+    var applicationLocale = defaultLocale
+        private set
+
+    /**
+     * Init this object
+     */
+    fun init(context: Context) {
+        val preferences = PreferenceManager.getDefaultSharedPreferences(context)
```

Comment:
could be great to let sharedPref being provided as a dependency, to improve testability